tests/settings/SectionTest.py

```diff
@@ -22,6 +23,16 @@ def test_append(self):
         self.assertEqual(str(uut.get('5 ')), '5')
         self.assertEqual(int(uut.get('nonexistent', 5)), 5)
 
+    def test_settings_override_warning(self):
+        logger = logging.getLogger()
+        with self.assertLogs(logger, 'WARNING') as cm:
+            uut = Section('name', None)
+            uut.add_or_create_setting(Setting('key1', 'value1'), None)
+            uut.add_or_create_setting(Setting('key1', 'value2'), None, False)
```

Comment:
> Maybe we should have tests for the situation when allow_appending is True and the user defines the setting multiple times

This has been done. Thanks ! However adding a test for custom keys would be irrelevant IMO.